metagraph/tests/test_utils.cpp

```diff
@@ -469,9 +469,16 @@ TEST(ThreadPool, MultiThreadException) {
             ThreadPool pool(i);
 
             std::vector<std::future<size_t>> result;
+            std::atomic<size_t> exception_count(0);
+            std::exception_ptr exception = nullptr;
             for (size_t t = 0; t < 1000; ++t) {
                 result.emplace_back(pool.enqueue([&](size_t i) {
-                    throw std::runtime_error("error");
+                    try {
+                        throw std::runtime_error("error in test");
+                    } catch (...) {
+                        if (!exception_count.fetch_add(1))
+                            exception = std::current_exception();
+                    }
```

Comment:
Hm... this test seems useless. It doesn't actually test the thread pool.

This code will never throw because you catch all exceptions and increment the counter.

So it the result of the test will never depend on the implementation of the thread pool and how exceptions are handled there.

I will try to propose a different test...

Reply:
It seems the only thing it makes sense to test here is that it throws in worker threads.

I test it by checking an implication of this -- that the main thread dies.